sources/packages/backend/apps/queue-consumers/src/processors/schedulers/sfas-integration/_tests_/sfas-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,191 @@
+import { DeepMocked, createMock } from "@golevelup/ts-jest";
+import { INestApplication } from "@nestjs/common";
+import { QueueNames, getISODateOnlyString } from "@sims/utilities";
+import {
+  createTestingAppModule,
+  describeProcessorRootTest,
+} from "../../../../../test/helpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  mockDownloadFiles,
+  saveFakeStudent,
+  saveFakeStudentRestriction,
+  SFAS_ALL_RESTRICTIONS_FILENAME,
+  SFAS_LEGACY_RESTRICTION_FILENAME,
+} from "@sims/test-utils";
+import * as Client from "ssh2-sftp-client";
+import { Job } from "bull";
+import * as path from "path";
+import { SFASIntegrationScheduler } from "../sfas-integration.scheduler";
+import { In } from "typeorm";
+import { RestrictionCode } from "@sims/services";
+
+describe(describeProcessorRootTest(QueueNames.SFASIntegration), () => {
+  let app: INestApplication;
+  let processor: SFASIntegrationScheduler;
+  let db: E2EDataSources;
+  let sftpClientMock: DeepMocked<Client>;
+  let sfasDownloadFolder: string;
+
+  beforeAll(async () => {
+    sfasDownloadFolder = path.join(__dirname, "sfas-files");
+    // Set the SFAS folder to the files mocks folders.
+    process.env.SFAS_RECEIVE_FOLDER = sfasDownloadFolder;
+    const { nestApplication, dataSource, sshClientMock } =
+      await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+    sftpClientMock = sshClientMock;
+    // Processor under test.
+    processor = app.get(SFASIntegrationScheduler);
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    // Empty the student restrictions table for each test.
+    const studentRestrictionIds = await db.studentRestriction.find({
+      select: { id: true },
+    });
+    const ids = studentRestrictionIds.map(
+      (studentRestrictionId) => studentRestrictionId.id,
+    );
+    await db.studentRestriction.delete({ id: In(ids) });
+  });
+
+  it("Should save only one legacy restriction for the student if one or more restrictions are imported from SFAS which do not have a mapping to the SIMS restrictions", async () => {
```

Comment:
We follow a comment pattern of `//Arrange` `//Act` `//Assert` to our tests and Arrange is missing in both the tests.